### PR TITLE
Set default fixed point decimal places to 18

### PR DESCRIPTION
**Pull request by randomnetcat**
Update default decimal places to 18
Update documentation

### CODE REVIEW

File: docs/types.rst [hunk 1]
```diff
@@ -75,19 +75,22 @@ sign extends. Shifting by a negative amount throws a runtime exception.
...
 the type and ``N`` represents how many decimal points are available. ``M`` must be divisible by 8 and goes from 8 to 256 bits. ``N`` must be between 0 and 80, inclusive.
-``ufixed`` and ``fixed`` are aliases for ``ufixed128x19`` and ``fixed128x19``, respectively.
+``ufixed`` and ``fixed`` are aliases for ``ufixed128x18`` and ``fixed128x18``, respectively.
 
 Operators:
 
 * Comparisons: ``<=``, ``<``, ``==``, ``!=``, ``>=``, ``>`` (evaluate to ``bool``)
 * Arithmetic operators: ``+``, ``-``, unary ``-``, unary ``+``, ``*``, ``/``, ``%`` (remainder)
 
+Conversions:
```
Comment (axic) [inline docs/types.rst]: Oh, this change needs to stay in the old PR.

File: docs/types.rst [hunk 2]
```diff
@@ -75,19 +75,22 @@ sign extends. Shifting by a negative amount throws a runtime exception.
 Fixed Point Numbers
 -------------------
 
-.. warning::
```
Comment (axic) [inline docs/types.rst]: This needs to stay here.

**Review by axic — CHANGES_REQUESTED** [review]
Please also pull in the test changes or the 19 -> 18 move.

**Comment by randomnetcat** [comment]
The tests won't pass without the rest of the fixed point code, regardless of decimal places.

**Comment by axic** [comment]
> The tests won't pass without the rest of the fixed point code, regardless of decimal places.

There are tests which expect a certain error message, now they read `fixed128x19`, but after the change here they must be adjusted, just as you did in the other PR.

**Comment by randomnetcat** [comment]
OH, sorry. Will fix those.

File: cmake-build-debug/CMakeFiles/3.10.2/CMakeCCompiler.cmake [hunk 3]
```diff
@@ -0,0 +1,73 @@
+set(CMAKE_C_COMPILER "/usr/bin/cc")
```
Comment (axic) [inline cmake-build-debug/CMakeFiles/3.10.2/CMakeCCompiler.cmake]: I think you've added an extra directory by mistake. I'll just force push myself.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
Test failure is due to #3802.

**Review by chriseth — APPROVED** [review]

**Pull request merged**